ingestion/functions/common/parsing_lib.py

```diff
@@ -155,12 +158,21 @@ def write_to_server(
         if not batch:
             break
         total_wait = 0
+        connection_tries = 0
         wait = 10  # initial wait time in seconds
         print(f"Sending {len(batch)} cases, total so far: {counter['total']}")
         # Exponential backoff in dev and prod, but not for local testing
-        while total_wait <= (MAX_WAIT_TIME if env in ["dev", "prod"] else 0):
-            res = requests.post(put_api_url, json={"cases": batch},
-                                headers=headers, cookies=cookies)
+        while (
+            total_wait <= (MAX_WAIT_TIME if env in ["dev", "prod"] else 0)
+            and connection_tries < MAX_CONN_TRIES
+        ):
+            try:
+                res = requests.post(put_api_url, json={"cases": batch},
+                                    headers=headers, cookies=cookies)
+            except requests.exceptions.ConnectionError:
+                print(f"Failed to connect to data service, waiting {CONN_WAIT}s")
+                connection_tries += 1
+                time.sleep(CONN_WAIT)
```

Comment:
if this fails all MAX_CONN_TRIES times we probably want an alarm because the data service has disappeared.